src/odemis/gui/comp/overlay/ellipse.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+"""
+:created: 2024-02-02
+:author: Nandish Patel
+:copyright: © 2024 Nandish Patel, Delmic
+
+This file is part of Odemis.
+
+.. license::
+    Odemis is free software: you can redistribute it and/or modify it under the
+    terms of the GNU General Public License version 2 as published by the Free
+    Software Foundation.
+
+    Odemis is distributed in the hope that it will be useful, but WITHOUT ANY
+    WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+    PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along with
+    Odemis. If not, see http://www.gnu.org/licenses/.
+
+"""
+import math
+
+import cairo
+
+from odemis import model, util
+import odemis.util.units as units
+import odemis.gui as gui
+from odemis.gui.comp.overlay.base import Vec
+from odemis.gui.comp.overlay.rectangle import RectangleOverlay
+
+NUM_ARCS_FACTOR = 3
+
+
+class EllipseOverlay(RectangleOverlay):
+    """Overlay representing one ellipse."""
+
+    def __init__(self, cnvs, colour=gui.SELECTION_COLOUR):
+        """
+        cnvs: canvas for the overlay
+        colour (str): hex colour code for the ellipse
+        """
+        super().__init__(cnvs, colour)
+        self.points = model.ListVA()
```

Comment:
To me, it's really not clear what's the different between .point and .coordinates. Also the content is not clear. It seems polygon and eclipses have .points, while rectangles have .coordinates. Why can't all the shapes have the same thing? rectangles are also a type of polygon.